src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -846,14 +851,14 @@ otError MeshForwarder::LogMeshFragmentHeader(MessageAction       aAction,
     bool                   shouldLogRss;
     Lowpan::MeshHeader     meshHeader;
     Lowpan::FragmentHeader fragmentHeader;
+    uint16_t               meshHeaderLength;
 
-    SuccessOrExit(meshHeader.Init(aMessage));
-    VerifyOrExit(meshHeader.IsMeshHeader());
+    SuccessOrExit(meshHeader.ParseFrom(aMessage, /* aOffset */ 0, meshHeaderLength));
```

Comment:
Is `MeshHeader::ParseFrom` ever called with `aOffset != 0`?

Reply:
Not currently (and I don't expect we would need it).
We can keep it simple (as it was) and assume we read MeshHeader at zero offset.